lib/matplotlib/axes/_axes.py

```diff
@@ -725,6 +725,7 @@ def axhline(self, y=0, xmin=0, xmax=1, **kwargs):
         trans = self.get_yaxis_transform(which='grid')
         l = mlines.Line2D([xmin, xmax], [y, y], transform=trans, **kwargs)
         self.add_line(l)
+        self.ignore_existing_data_limits = True
```

Comment:
I have made a similar change to `axvline`. Please see my comment below approving the changes.